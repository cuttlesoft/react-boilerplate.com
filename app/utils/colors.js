import { lighten, darken } from 'polished'

/**
 * Takes an array of colors and adds them to the colors object with numbered prefixes
 *
 * Example Input:
 * const colors = {}
 * const demoColors = ['#000', '#333', '#666']
 *
 * Example Output:
 * colorArray(demoColors, 'demo')
 * -> { 'demo-1': '#000', 'demo-2': '#333', 'demo-3': '#666', } // colors
 */
const colorArray = (array, prefix) =>
  array.forEach((color, index) => {
    colors[`${prefix}-${index + 1}`] = color
  })

export const baseColors = {
  red: 'rgb(230,34,45)',
  pink: 'rgb(221,17,117)',
  orange: 'rgb(245,186,23)',
  yellow: 'rgb(242,233,0)',
  green: 'rgb(112, 189, 70)',
  blue: 'rgb(24,192,221)',
  purple: 'rgb(65,86,166)',
  darkGrey: 'rgb(68,68,68)',
  mediumGrey: 'rgb(170,170,170)',
  lightGrey: 'rgb(238,238,238)',
  white: 'rgb(255,255,255)',
  transparent: '#00000000',
}

/*
 * Add helper colors to allow for more flexible styling
 * Primary will map to Grommet's `brand`
 * Secondary will map to Grommet's `focus`
 */
baseColors.primary = baseColors.blue
baseColors.secondary = baseColors.purple
baseColors.primaryText = baseColors.darkGrey
baseColors.background = baseColors.white

/**
 * Dark and light colors are based on the primaryText and lightGrey colors, respectively.
 * This allows us to swap out a dark theme and light theme while still being able to use
 * theme colors such as `dark-2` or `light-4`.
 */
const darkColors = [
  darken(0.07, baseColors.primaryText),
  darken(0.05, baseColors.primaryText),
  darken(0.025, baseColors.primaryText),
  baseColors.primaryText,
  lighten(0.025, baseColors.primaryText),
  lighten(0.05, baseColors.primaryText),
]
const lightColors = [
  lighten(0.02, baseColors.lightGrey),
  baseColors.lightGrey,
  darken(0.2, baseColors.lightGrey),
  darken(0.3, baseColors.lightGrey),
  darken(0.4, baseColors.lightGrey),
  darken(0.5, baseColors.lightGrey),
]

/** @todo customize these to match theme colors! */
const accentColors = [baseColors.pink, baseColors.green, '#60EBE1', '#FFCA58']
const neutralColors = [
  lighten(0.2, baseColors.pink),
  lighten(0.2, baseColors.green),
  '#6095EB',
  '#FFB200',
]
const statusColors = {
  critical: '#FF3333',
  error: '#FF3333',
  warning: baseColors.yellow,
  ok: baseColors.green,
  unknown: baseColors.mediumGrey,
  disabled: baseColors.mediumGrey,
}

/*
 * Base theme colors, passed directly as theme colors
 * The keys used here should be found in;
 * https://github.com/grommet/grommet/blob/b70077fe02311aaf5e6b1a9f256077982daf9249/src/js/themes/base.js#L53
 */
export const colors = {
  active: baseColors.mediumGrey,
  background: baseColors.background,
  brand: baseColors.primary,
  control: {
    dark: baseColors.primary,
    light: baseColors.secondary,
  },
  focus: baseColors.secondary,
  icon: {
    dark: '#f8f8f8',
    light: '#666666',
  },
  placeholder: '#AAAAAA',
  text: {
    dark: '#eeeeee',
    light: '#444444',
  },
  white: '#FFFFFF',
}
// Add custom color array to colors
colorArray(accentColors, 'accent')
colorArray(neutralColors, 'neutral')
colorArray(darkColors, 'dark')
colorArray(lightColors, 'light')
Object.keys(statusColors).forEach(color => {
  colors[`status-${color}`] = statusColors[color]
})

export default colors /** @todo replace with named import for files that use this */
