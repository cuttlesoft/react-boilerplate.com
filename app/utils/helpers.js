/* istanbul ignore next */

const helpers = {
  createWithDoc({ envName = '', docFunction = () => {}, component = '' }) {
    let createComponentWithDoc
    if (envName !== 'production') {
      createComponentWithDoc = docFunction(component) // eslint-disable-line global-require
    }
    return createComponentWithDoc || component
  },
}

export default helpers
