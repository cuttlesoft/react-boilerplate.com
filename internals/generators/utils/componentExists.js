/**
 * componentExists
 *
 * Check whether the given component exist in either the components or pages directory
 */

const fs = require('fs');
const path = require('path');
const pageComponents = fs.readdirSync(
  path.join(__dirname, '../../../app/components'),
);
const pagepages = fs.readdirSync(
  path.join(__dirname, '../../../app/pages'),
);
const components = pageComponents.concat(pagepages);

function componentExists(comp) {
  return components.indexOf(comp) >= 0;
}

module.exports = componentExists;
