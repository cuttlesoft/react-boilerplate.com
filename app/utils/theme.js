import { css } from 'styled-components'

import { baseColors, colors } from './colors'

export const HEADER_FONT = 'soleil'
export const BODY_FONT = 'Open Sans'

const theme = {
  global: {
    colors,
    drop: {
      background: colors.background,
      extend: `box-shadow: 0px 4px 4px ${colors['dark-1']}; font-weight: bold`,
    },
    font: {
      family: BODY_FONT,
      size: 14,
    },
    input: {
      weight: 700,
    },
  },
  anchor: {
    color: 'control',
  },
  button: {
    border: {
      radius: '4px',
    },
  },
  layer: {
    background: colors.background,
    overlay: {
      background: 'rgba(48, 48, 48, 0.5)',
    },
  },
  select: {
    background: colors.background,
    control: {
      extend: 'border-radius: 4px; border: 1px solid #4e504c',
    },
  },
  tabs: {
    gap: 'medium',
  },
  tab: {
    active: {
      color: 'white',
    },
    color: colors['light-5'],
    border: {
      side: 'bottom',
      size: 'small',
      color: {
        dark: colors['dark-1'],
        light: baseColors.transparent,
      },
      active: {
        color: {
          dark: baseColors.primary,
          light: baseColors.primary,
        },
      },
      hover: {
        color: {
          dark: colors.brand,
          light: colors.brand,
        },
      },
    },
    extend: () => css`
      font-weight: bold;
    `,
  },
  table: {
    header: 'none',
    body: {
      pad: 'none',
    },
  },
}

export default theme
