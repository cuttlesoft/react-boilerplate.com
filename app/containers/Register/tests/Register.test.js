import React from 'react'
import { render } from '@testing-library/react'
// import '@testing-library/jest-dom/extend-expect' // add some helpful assertions

import { TestWrapper } from 'utils/TestWrapper'
import { Register } from '../index'

const renderComponent = (props = {}) =>
  render(
    <TestWrapper>
      <Register {...props} />{' '}
    </TestWrapper>,
  )

/**
 *
 * Tests for Register
 *
 */
describe('Register', () => {
  it('renders and matches snapshot', () => {
    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })
})
