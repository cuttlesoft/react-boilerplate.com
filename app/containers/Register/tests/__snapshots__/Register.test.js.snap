// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register renders and matches snapshot 1`] = `
<div>
  <div
    class="StyledGrommet-sc-19lkkz7-0 eTWpSM"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 QJUym"
    >
      <img
        class="StyledImage-ey4zx9-0 cHpGKY"
        src="IMAGE_MOCK"
        style="width: 300px;"
      />
    </div>
    <div
      class="StyledBox-sc-13pk1d4-0 ewTzDI"
      style="width: 90vw; max-width: 600px;"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 gVSbfC"
      >
        <h2
          class="StyledHeading-sc-1rdh4aw-0 fjsCgo"
          style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
        >
          Create an account
        </h2>
        <a
          class="Link__StyledLink-sc-1vpl4pu-0 gEErvb"
          href="/login"
          theme="[object Object]"
        >
          or Sign In
        </a>
      </div>
      <form
        id="register-form"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 kEDHqi"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 bIVpEL FormField__FormFieldBox-m9hood-0 ddgSii FormField__StyledFormField-g4wt0p-0 feezXd"
            style="flex: 1;"
          >
            <label
              class="StyledText-sc-1sadyjn-0 enbIRN"
            >
              <span
                class="StyledText-sc-1sadyjn-0 lmFPCJ Text__StyledGrommetText-sc-1qclgcc-0 lkunaY"
              >
                First Name
                <span
                  class="StyledText-sc-1sadyjn-0 hnHDhw Text__StyledGrommetText-sc-1qclgcc-0 lkunaY"
                >
                  *
                </span>
              </span>
            </label>
            <div
              class="StyledBox-sc-13pk1d4-0 GCqjV"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 jJdKAu"
              >
                <div
                  class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 ksQrmr"
                >
                  <input
                    autocomplete="off"
                    class="StyledTextInput-sc-1x30a0s-0 ickzDR"
                    name="first_name"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffjnxm"
          />
          <div
            class="StyledBox-sc-13pk1d4-0 bIVpEL FormField__FormFieldBox-m9hood-0 ddgSii FormField__StyledFormField-g4wt0p-0 feezXd"
            style="flex: 1;"
          >
            <label
              class="StyledText-sc-1sadyjn-0 enbIRN"
            >
              <span
                class="StyledText-sc-1sadyjn-0 lmFPCJ Text__StyledGrommetText-sc-1qclgcc-0 lkunaY"
              >
                Last Name
                <span
                  class="StyledText-sc-1sadyjn-0 hnHDhw Text__StyledGrommetText-sc-1qclgcc-0 lkunaY"
                >
                  *
                </span>
              </span>
            </label>
            <div
              class="StyledBox-sc-13pk1d4-0 GCqjV"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 jJdKAu"
              >
                <div
                  class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 ksQrmr"
                >
                  <input
                    autocomplete="off"
                    class="StyledTextInput-sc-1x30a0s-0 ickzDR"
                    name="last_name"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 bIVpEL FormField__FormFieldBox-m9hood-0 ddgSii FormField__StyledFormField-g4wt0p-0 feezXd"
        >
          <label
            class="StyledText-sc-1sadyjn-0 enbIRN"
          >
            <span
              class="StyledText-sc-1sadyjn-0 lmFPCJ Text__StyledGrommetText-sc-1qclgcc-0 lkunaY"
            >
              Email
              <span
                class="StyledText-sc-1sadyjn-0 hnHDhw Text__StyledGrommetText-sc-1qclgcc-0 lkunaY"
              >
                *
              </span>
            </span>
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 GCqjV"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 jJdKAu"
            >
              <div
                class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 ksQrmr"
              >
                <input
                  autocomplete="off"
                  class="StyledTextInput-sc-1x30a0s-0 gknGre"
                  name="email"
                  type="email"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 bIVpEL FormField__FormFieldBox-m9hood-0 ddgSii FormField__StyledFormField-g4wt0p-0 feezXd"
        >
          <label
            class="StyledText-sc-1sadyjn-0 enbIRN"
          >
            <span
              class="StyledText-sc-1sadyjn-0 lmFPCJ Text__StyledGrommetText-sc-1qclgcc-0 lkunaY"
            >
              Password
              <span
                class="StyledText-sc-1sadyjn-0 hnHDhw Text__StyledGrommetText-sc-1qclgcc-0 lkunaY"
              >
                *
              </span>
            </span>
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 GCqjV"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 jJdKAu"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 emmXTg"
              >
                <div
                  class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 ksQrmr"
                >
                  <input
                    autocomplete="off"
                    class="StyledTextInput-sc-1x30a0s-0 gknGre"
                    id="password"
                    name="password"
                    type="password"
                    value=""
                  />
                </div>
                <button
                  class="StyledButton-sc-323bzc-0 jvkqEN Button__StyledGrommetButton-sc-5cxguj-0 liIZVq"
                  type="button"
                >
                  <svg
                    aria-label="Hide"
                    class="StyledIcon-ofa7kd-0 dWXdkj"
                    style="width: 30px;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12,17 C9.27272727,17 6,14.2222222 6,12 C6,9.77777778 9.27272727,7 12,7 C14.7272727,7 18,9.77777778 18,12 C18,14.2222222 14.7272727,17 12,17 Z M11,12 C11,12.55225 11.44775,13 12,13 C12.55225,13 13,12.55225 13,12 C13,11.44775 12.55225,11 12,11 C11.44775,11 11,11.44775 11,12 Z M20,5 L4,19"
                      fill="none"
                      stroke="#000"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 bIVpEL FormField__FormFieldBox-m9hood-0 ddgSii FormField__StyledFormField-g4wt0p-0 feezXd"
        >
          <label
            class="StyledText-sc-1sadyjn-0 enbIRN"
          >
            <span
              class="StyledText-sc-1sadyjn-0 lmFPCJ Text__StyledGrommetText-sc-1qclgcc-0 lkunaY"
            >
              Re-enter Password
              <span
                class="StyledText-sc-1sadyjn-0 hnHDhw Text__StyledGrommetText-sc-1qclgcc-0 lkunaY"
              >
                *
              </span>
            </span>
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 GCqjV"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 jJdKAu"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 emmXTg"
              >
                <div
                  class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 ksQrmr"
                >
                  <input
                    autocomplete="off"
                    class="StyledTextInput-sc-1x30a0s-0 gknGre"
                    name="confirm_password"
                    type="password"
                    value=""
                  />
                </div>
                <button
                  class="StyledButton-sc-323bzc-0 jvkqEN Button__StyledGrommetButton-sc-5cxguj-0 liIZVq"
                  type="button"
                >
                  <svg
                    aria-label="Hide"
                    class="StyledIcon-ofa7kd-0 dWXdkj"
                    style="width: 30px;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12,17 C9.27272727,17 6,14.2222222 6,12 C6,9.77777778 9.27272727,7 12,7 C14.7272727,7 18,9.77777778 18,12 C18,14.2222222 14.7272727,17 12,17 Z M11,12 C11,12.55225 11.44775,13 12,13 C12.55225,13 13,12.55225 13,12 C13,11.44775 12.55225,11 12,11 C11.44775,11 11,11.44775 11,12 Z M20,5 L4,19"
                      fill="none"
                      stroke="#000"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 bIVpEL FormField__FormFieldBox-m9hood-0 ddgSii FormField__StyledFormField-g4wt0p-0 feezXd"
        >
          <label
            class="StyledText-sc-1sadyjn-0 enbIRN"
          >
            <span
              class="StyledText-sc-1sadyjn-0 lmFPCJ Text__StyledGrommetText-sc-1qclgcc-0 lkunaY"
            >
              Account Type.
              <span
                class="StyledText-sc-1sadyjn-0 hnHDhw Text__StyledGrommetText-sc-1qclgcc-0 lkunaY"
              >
                *
              </span>
            </span>
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 GCqjV"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 eXTSqT"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 eitMOG"
              >
                <label
                  class="StyledRadioButton__StyledRadioButtonContainer-g1f6ld-0 kKDEPl"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 jJdKAu StyledRadioButton-g1f6ld-4 eLBYmH"
                  >
                    <input
                      class="StyledRadioButton__StyledRadioButtonInput-g1f6ld-1 hBCEfd"
                      name="role"
                      type="radio"
                      value="Supplier Admin"
                    />
                    <div
                      class="StyledBox-sc-13pk1d4-0 fmmWcA"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 lmFPCJ Text__StyledGrommetText-sc-1qclgcc-0 lkunaY"
                      >
                        Supplier
                      </span>
                    </div>
                  </div>
                </label>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cPZcyD"
                />
                <label
                  class="StyledRadioButton__StyledRadioButtonContainer-g1f6ld-0 kKDEPl"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 jJdKAu StyledRadioButton-g1f6ld-4 eLBYmH"
                  >
                    <input
                      class="StyledRadioButton__StyledRadioButtonInput-g1f6ld-1 hBCEfd"
                      name="role"
                      type="radio"
                      value="Broker Admin"
                    />
                    <div
                      class="StyledBox-sc-13pk1d4-0 fmmWcA"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 lmFPCJ Text__StyledGrommetText-sc-1qclgcc-0 lkunaY"
                      >
                        Broker
                      </span>
                    </div>
                  </div>
                </label>
              </div>
            </div>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 tjSEQ"
        >
          <label
            class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 PHOTu"
            for="check-box"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 jmffUi StyledCheckBox-sc-1dbk5ju-6 CBoMo"
            >
              <input
                class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 dRYNyB"
                id="check-box"
                type="checkbox"
              />
              <div
                class="StyledBox-sc-13pk1d4-0 uqlPU StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 iEciRS"
              >
                
              </div>
            </div>
          </label>
          <span
            class="StyledText-sc-1sadyjn-0 itJhXE Text__StyledGrommetText-sc-1qclgcc-0 lkunaY"
          >
            I agree to the
             
            <span
              class="StyledText-sc-1sadyjn-0 gEmiHW Text__StyledGrommetText-sc-1qclgcc-0 lkunaY"
            >
               Terms of Service, Privacy Policy and Cookie Policy
            </span>
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hzTjkL"
        >
          <button
            class="StyledButton-sc-323bzc-0 cGKVeS Button__StyledGrommetButton-sc-5cxguj-0 liIZVq"
            data-testid="submit-button"
            type="submit"
          >
            Register
          </button>
        </div>
      </form>
    </div>
     
  </div>
</div>
`;
